Skipper: init at 3.2.25.1701

NicolasGuilloux opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change

Skipper: init at 3.2.25.1701 (fixes partially #99232).

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

NicolasGuilloux commented:
_TODO:_
- [ ] Understand why it tries to read into `/var/log/installer/syslog`
- [ ] Find out why it can't save the registration
- [ ] Test skipper
- [ ] Clean the git history
- [ ] Maybe clean some useless dependency in Pulpo
- [ ] Checks that everything fits with the Contributing requirements

Since it is my first package, I would greatly appreciate some feedback :) Cheers guys !

NicolasGuilloux commented:
I cleaned everything. If anyone wants to test or merge it, it is good to go! ;)

I still have that annoying attempt to access to `/var/log/installer/syslog` without success, but it does not harm the stability of the software.

NicolasGuilloux commented:
Do I have to bind somebody about this PR? I think like the PR is ready to be merged.

SuperSandro2000 commented on maintainers/maintainer-list.nix:
```diff
@@ -6237,6 +6237,12 @@
+  nover = {
+    email = "novares.x@gmail.com";
+    github = "NicolasGuilloux";
+    githubId = 4090627;
```
Please format the commit message like ``maintainers: add nover``

SuperSandro2000 commented on pkgs/applications/editors/inventic/skipper.nix:
```diff
@@ -0,0 +1,65 @@
+{ stdenv, lib, fetchurl, autoPatchelfHook, makeWrapper, unzip,
+
+# Unpack libs
```
```suggestion
```

SuperSandro2000 commented on pkgs/applications/editors/inventic/skipper.nix:
```diff
@@ -0,0 +1,65 @@
+boost172, curl, libX11, libpng12, libxml2, libxslt, openssl, postgresql, qt5
+, zlib }:
+
+with lib;
```
Please only use it where necessary in a small scope.

SuperSandro2000 commented on pkgs/applications/editors/inventic/skipper.nix:
```diff
@@ -0,0 +1,65 @@
+  pname = "skipper";
+  version = "3.2.25.1701";
+  src = fetchurl {
+    url =
+      "https://downloads.skipper18.com/${version}/Skipper-${version}-Linux-all-64bit.zip";
```
```suggestion
    url = "https://downloads.skipper18.com/${version}/Skipper-${version}-Linux-all-64bit.zip";
```

SuperSandro2000 commented on pkgs/applications/editors/inventic/skipper.nix:
```diff
@@ -0,0 +1,65 @@
+    postgresql
+    qt5.qtbase
+    qt5.qtscript
+    stdenv.cc.cc.lib
```
That looks strange to me and so far I only have seen it when doing cross compile.

SuperSandro2000 commented on pkgs/applications/editors/inventic/skipper.nix:
```diff
@@ -0,0 +1,65 @@
+  runtimeDependencies = [ libpng12 ];
+
+  unpackPhase = ''
+    unzip $src
```
fetchzip should do this automatically.

SuperSandro2000 commented on pkgs/applications/editors/inventic/skipper.nix:
```diff
@@ -0,0 +1,65 @@
+  '';
+
+  meta = with stdenv.lib; {
+    description =
+      "An ORM framework editor for Doctrine, Symfony, Laravel, etc.";
```
```suggestion
    description = "An ORM framework editor for Doctrine, Symfony, Laravel, etc.";
```

NicolasGuilloux replied:
I removed nixfmt, apparently it is not that good ;)

SuperSandro2000 commented on pkgs/applications/editors/inventic/skipper.nix:
```diff
@@ -0,0 +1,65 @@
+  '';
+
+  postFixup = ''
+    wrapQtApp $out/opt/Skipper
```
Please avoid double wrapping.

NicolasGuilloux replied:
I tried to remote it, but when I start the package, it fails to launch with the following error:

```
##WARNING: Library libxcb-xinerama.so.0 is missing on your system. Please install it by using 'sudo apt-get install libxcb-xinerama0'
qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

[1]    28569 abort (core dumped)  skipper
```

With the postFixup, everything works well. Do you know why?

SuperSandro2000 replied:
wrapQtApp uses internally makeWrapper and we can pass the arguments it uses to the same wrapper that already sets LD_LIBRARY_PATH.

SuperSandro2000 commented on pkgs/applications/editors/inventic/skipper.nix:
```diff
@@ -0,0 +1,65 @@
+    description =
+      "An ORM framework editor for Doctrine, Symfony, Laravel, etc.";
+    homepage = "https://www.skipper18.com/";
+    license = [ licenses.unfree ];
```
```suggestion
    license = licenses.unfree;
```

SuperSandro2000 commented on pkgs/applications/editors/inventic/skipper.nix:
```diff
@@ -0,0 +1,65 @@
+    homepage = "https://www.skipper18.com/";
+    license = [ licenses.unfree ];
+    maintainers = with maintainers; [ nover ];
+    platforms = platforms.linux;
```
```suggestion
    platforms = [ "x86_64-linux" ];
```

The binary won't work on i686.

SuperSandro2000 commented:
Result of `nixpkgs-review pr 99233` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>skipper</li>
  </ul>
</details>

NicolasGuilloux committed:
maintainers: add nover

NicolasGuilloux commented:
Hi @SuperSandro2000 

Sorry for the delay. I was pretty busy and I completely forgot to fix your feedbacks :(
Anyway, I fixed what you asked, except the `postFixup`. I don't know why it fails if I remove it. I though the hook added in the `nativeBuildInputs` would take care of that.

Anyway, about the package itself, I added a `.desktop` entry. Tell me if there is a better way of doing it.
I also updated the package to the latest version.

Cheers! :D

SuperSandro2000 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -20944,6 +20944,8 @@ in
 
   rhvoice = callPackage ../applications/audio/rhvoice { };
 
+  skipper = callPackage ../applications/editors/inventic/skipper.nix {};
```
```suggestion
  skipper = callPackage ../applications/editors/inventic/skipper.nix { };
```

SuperSandro2000 commented on pkgs/applications/editors/inventic/skipper.nix:
```diff
@@ -0,0 +1,64 @@
+    wrapQtApp $out/opt/Skipper
+  '';
+
+  meta = with stdenv.lib; {
```
```suggestion
  meta = with lib; {
```

SuperSandro2000 commented on pkgs/applications/editors/inventic/skipper.nix:
```diff
@@ -0,0 +1,64 @@
+      --replace "Icon=" "Icon=$out/opt/Skipper.ico"
+  '';
+
+  postFixup = ''
+    wrapQtApp $out/opt/Skipper
+  '';
```
```suggestion
```

SuperSandro2000 commented on pkgs/applications/editors/inventic/skipper.nix:
```diff
@@ -0,0 +1,64 @@
+    mv ./* $out/opt
+
+    makeWrapper $out/opt/Skipper $out/bin/skipper \
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath runtimeDependencies}
```
```suggestion
      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath runtimeDependencies} \
      ''${qtWrapperArgs[@]}
```

NicolasGuilloux committed:
skipper: init at 3.2.31.1730

NicolasGuilloux commented:
I don't know if I can ping you @SuperSandro2000, sorry for the spam :(

SuperSandro2000 commented:
Do you need help with something? I am just a bit behind my github notifications and will catch up with them in the coming days.

SuperSandro2000 commented on pkgs/applications/editors/inventic/skipper.nix:
```diff
@@ -0,0 +1,60 @@
+, zlib }:
+
+let
+  desktopFileTemplate = ./skipper.desktop;
```
We have a function called makeDesktopItem which should be used instead of this extra file.

SuperSandro2000 commented on pkgs/applications/editors/inventic/skipper.nix:
```diff
@@ -0,0 +1,60 @@
+
+    mkdir -p $out/opt
+    chmod +x ./Skipper
+    mv ./* $out/opt
```
nixpkgs has not directory named opt in $out. Please move binaries to $out/bin.

Janik-Haag reviewed (commented):
@NicolasGuilloux are you still interested in getting this merged? I will close it in a few wekks if you aren't.

NicolasGuilloux commented:
Hey! No we can close it since I stopped using Skipper, I'm don't have any license and therefore cannot test. 
Thanks for reminding me that this MR was sleeping though 😉

Janik-Haag commented:
No problem ^^
